litestar/connection/base.py

```diff
@@ -234,12 +234,12 @@ def session(self) -> dict[str, Any]:
         Raises:
             ImproperlyConfiguredException: if session is not set in scope.
         """
-        if "session" not in self.scope:
+        if SCOPE_SESSION_KEY not in self.scope or self.scope[SCOPE_SESSION_KEY] is Empty:
             raise ImproperlyConfiguredException(
                 "'session' is not defined in scope, install a SessionMiddleware to set it"
             )
 
-        return cast("dict[str, Any]", self.scope["session"])
+        return cast("dict[str, Any]", self.scope[SCOPE_SESSION_KEY])
```

Comment:
This `cast()` hides quite a bit more, it seems.

`BaseScope.session` is typed as `EmptyType | Dict[str, Any] | DataContainerType | None`, where `DataContainerType` is `Struct | BaseModel | AttrsInstance | TypedDictClass | DataclassProtocol`.

I'm not sure where the scope might be swapped out for one of the model types like Struct et al., but it seems as though type checkers won't prevent us from letting that happen.. nor will they prevent this property from returning `None`.

Reply:
Hm. I'm wondering if we might want to change the scope type?

Reply:
Or, we _could_ just type it correctly:

```py
    @property
    def session(self) -> ScopeSession:
```

It throws a stack of type errors but only one of those is outside of the test suite.

I'm pretty green when it comes to session related aspects of the code base, so I think I'll take a bit more time to understand the problem before pulling the trigger on anything here.